homeassistant/components/evohome/manifest.json

```diff
@@ -3,7 +3,8 @@
   "name": "Evohome",
   "documentation": "https://www.home-assistant.io/components/evohome",
   "requirements": [
-    "evohomeclient==0.3.3"
+    "evohomeclient==0.3.3",
+    "python-dateutil==2.8.0"
```

Comment:
@pvizeli said: https://github.com/home-assistant/home-assistant/pull/25010#discussion_r300872940
> remove this and use our timezone function from utils

I responded: 
As I indicated [above](https://github.com/home-assistant/home-assistant/pull/25010#issue-295108679), (and [here](https://github.com/home-assistant/home-assistant/pull/24491#discussion_r299011440)) I could not work out how to get the functionality I required from `utils.dt` (nor from `pytz`).

What I need to do is determine the offset used by the client library (i.e. used by the host system), which uses local/naive `datetime` objects.

In  `utils.dt`, I _can_ `get_time_zone()` for a given TZ string (e.g. `"Europe\London"`): 
```python
def get_time_zone(time_zone_str: str) -> Optional[dt.tzinfo]:
```
But how would I know right zone name, and what about daylight saving times?

I can also get the TZ for an aware `datetime` using`dt.tzinfo`. But my `datetime` objects are naive.

So I use `dateutil.tzlocal()`:
```python
def _local_dt_to_utc(dt_naive: datetime) -> datetime:
    dt_aware = as_utc(dt_naive.replace(microsecond=0, tzinfo=tzlocal()))
    return dt_aware.replace(tzinfo=None)
```

Reply:
@balloob Can I proceed with using `dateutil.tzlocal()`? 

I can see no other way of achieving this functionality, unless the user is obligated to setting `time_zone:` perfectly:
```yaml
homeassistant:
  name: Home
  unit_system: metric
# time_zone: UTC
```

... and I find that unsatisfactory - it i sunkind to the user, and may will result in issues being reported.

Reply:
Timezone is a mandatory config field

Reply:
err, no? In https://www.home-assistant.io/docs/configuration/basic/#time_zone, I have: "**time_zone**
(string) (Optional) Pick your time zone from the column TZ of Wikipedia’s list of tz database time zones"

Reply:
If I am reading that right, and even if it was _required_ to be configured, I think that expecting the user to match it to their system's actual TZ (and maintain it as matched) is unnecessarily onerous, and a mistake in any case. Especially not when you can just go and look with `dateutil.dt.tzlocal()` and get the truth!

If we do best practice and store all TZs as UTC then what is `time_zone:` for? I would say its the TZ used when displaying datetimes in the UI.

Any client library that is providing naive `datetime`s doesn't have to match the configured `time_zone:` (they have no way of knowing it) - they will just use the system's TZ. 

When HA's TZ, and the 'real' TZ are mis-matched, then all sorts of obscure issues may arise from naive datetimes - I fact, I would be throwing a `_LOGGER.warning()` whenever the configured TZ doesn't match the actual TZ.

In addition, there is a library, `tzlocal`, written specifically to add `get_localzone` to `pytz` for this reason, acknowledging this as a limitation of `pytz`). `dateutil.tz` is recommend by Python 3.7 as _the_ means of access to the Olsen DB; `pytz` was last mentioned for Python 3.5). 

So an argument exists fro switching from pytz to dateutil - but I am not pushing for that here.